src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs

```diff
@@ -219,22 +212,43 @@ private GatewayModel.CommittedTransactionInfo MapToGatewayAccountTransaction(Led
 
         return new GatewayModel.CommittedTransactionInfo(
             stateVersion: lt.StateVersion,
-            transactionStatus: status,
+            transactionStatus: MapTransactionStatus(lt.EngineReceipt.Status),
             payloadHashHex: payloadHashHex,
             intentHashHex: intentHashHex,
             feePaid: lt.FeePaid.HasValue ? new GatewayModel.TokenAmount(lt.FeePaid.Value.ToString(), _networkConfigurationProvider.GetWellKnownAddresses().Xrd) : null,
             confirmedAt: lt.RoundTimestamp,
-            errorMessage: lt.ErrorMessage
+            errorMessage: lt.EngineReceipt.ErrorMessage
         );
     }
 
-    private DetailsLookupResult MapToGatewayAccountTransactionWithDetails(UserLedgerTransaction ult, List<Entity> referencedEntities)
+    private DetailsLookupResult MapToGatewayAccountTransactionWithDetails(UserLedgerTransaction ult, List<Entity> referencedEntities, List<GatewayModel.TransactionCommittedDetailsRequest.OptInPropertiesEnum>? optInProperties)
     {
+        var includeFeeSummary = optInProperties?.Contains(GatewayModel.TransactionCommittedDetailsRequest.OptInPropertiesEnum.ReceiptFeeSummary);
+        var includeStateChanges = optInProperties?.Contains(GatewayModel.TransactionCommittedDetailsRequest.OptInPropertiesEnum.ReceiptStateChanges);
+        var includeRawPayload = optInProperties?.Contains(GatewayModel.TransactionCommittedDetailsRequest.OptInPropertiesEnum.RawPayload);
+        var includeEvents = optInProperties?.Contains(GatewayModel.TransactionCommittedDetailsRequest.OptInPropertiesEnum.ReceiptEvents);
+
         return new DetailsLookupResult(MapToGatewayAccountTransaction(ult), new GatewayModel.TransactionCommittedDetailsResponseDetails(
-            rawHex: ult.RawPayload.ToHex(),
-            receipt: new JRaw(ult.EngineReceipt),
+            rawPayload: includeRawPayload == true ? ult.RawPayload.ToHex() : null,
+            receipt: new GatewayModel.TransactionReceipt
+            {
+                ErrorMessage = ult.EngineReceipt.ErrorMessage,
+                Status = MapTransactionStatus(ult.EngineReceipt.Status),
+                Items = new JRaw(ult.EngineReceipt.Items),
+                FeeSummary = includeFeeSummary == true ? new JRaw(ult.EngineReceipt.FeeSummary) : null,
+                NextEpoch = ult.EngineReceipt.NextEpoch != null ? new JRaw(ult.EngineReceipt.NextEpoch) : null,
+                StateUpdates = includeStateChanges == true ? new JRaw(ult.EngineReceipt.StateUpdates) : null,
+                Events = includeEvents == true ? new JRaw(ult.EngineReceipt.Events) : null,
```

Comment:
Ideally we should not even fetch them from DB but guess this should be good-enough.,

Reply:
I can do that, but that would mean replacing that entity framework query with raw sql with dapper.
If you feel it's right time for that, happy to address that.

Reply:
I don't think we need to invest that much into this area at the moment.